pinot-core/src/main/java/org/apache/pinot/core/operator/filter/JsonMatchFilterOperator.java

```diff
@@ -87,4 +94,20 @@ public String toExplainString() {
     stringBuilder.append(",predicate:").append(_predicate.toString());
     return stringBuilder.append(')').toString();
   }
+
+  private void record(ImmutableRoaringBitmap bitmap) {
+    InvocationRecording recording = Tracing.activeRecording();
+    if (recording.isEnabled()) {
+      recording.setColumnName(_predicate.getLhs().getIdentifier());
+      recording.setFilter(FilterType.INDEX, describeJsonPredicate());
+      recording.setNumDocsMatchingAfterFilter(bitmap.getCardinality());
+    }
+  }
+
+  private String describeJsonPredicate() {
```

Comment:
Since we already have the column name recorded, simply put the `_predicate.getType()` (it is always `JSON_MATCH`)?

Reply:
I'll leave it this way for now (recording JSON_MATCH) because I want to progress, but capturing the parsed filter expression was IMO a lot more valuable than this.

Reply:
I see what I was missing before. You added multiple traces for the json predicate, one for each sub-predicate under the json expression.
To get the detailed info, we have to go into the index reader then. We can probably add that separately if needed